File: 1351-count-negative-numbers-in-a-sorted-matrix/1351-count-negative-numbers-in-a-sorted-matrix.py
class Solution:
    def countNegatives(self, grid: List[List[int]]) -> int:
        m = len(grid)
        n = len(grid[0])
        
        if grid[0][0] < 0:
            return n * m
        if grid[-1][-1] >= 0:
            return 0
        
        row = m - 1
        col = 0
        count = 0
        
        while row >= 0 and col < n:
            if grid[row][col] < 0:
                count += n - col
                row -= 1
            else:
                col += 1
                    
        return count
                    
                
        

File: 1351-count-negative-numbers-in-a-sorted-matrix/1351-count-negative-numbers-in-a-sorted-matrix.java
class Solution {
    public int countNegatives(int[][] matrix) {
        int counter = 0;
        int n = matrix.length;    // number of rows
        int m = matrix[0].length; // number of columns
        
        if (matrix[0][0] < 0) {
            return n * m;
        }
        
        if (matrix[n - 1][m - 1] > 0) {
            return 0;
        }
        int col = 0;
        int row = n - 1;

        while (row >= 0 && col < m) {
            if (matrix[row][col] < 0) {
                counter += m - col;
                row--;
            } else {
                col++;
            }
        }
        return counter;
    }
}